Repo JSON, Expanded Format, and ReadMe

kleineluka opened the pull request:
- Organise and expand `ReadMe`
- Add support for an (optional) `Icon URL` in mod information
- Add support for (optional) tags `content`, `dev`, `qol`, `lib`, and `mature` in mod information
- Add end point to get the whole repository in a single json, with in-lined `mod.json` content (`/repo.json`)

kleineluka commented:
OneHalf said it was okay to add a member blinkie so I hope this is ok! If not, you are free to only merge the first commit (or neither of course haha)

gBasil commented on src/pages/repo.json.ts:
```diff
@@ -0,0 +1,9 @@
+// imports
```
The comments here and on line 5 are self-explanatory and can be removed.

gBasil commented on README.md:
````diff
@@ -1,43 +1,62 @@
 	// The rest of the JSON
 }
 ```
+<br/>**And to use the repository as an API:**<br>
````
I feel like this should be moved down or put elsewhere (maybe under a new section?), since otherwise it feels like it could be confusing for mod devs.

gBasil commented:
> - Add support for an (optional) Icon URL in mod information

I feel like that would be better suited to be added to the `mod.json` spec.

> OneHalf said it was okay to add a member blinkie so I hope this is ok!

I'm alright with it too :)

kleineluka commented:
Pushed small commit.

- followed your commenting standards and removed in repo.json.ts
- moved repository api to its own readme section
- removed icon as option 

Should be all good now. Not sure what to do about the icon other than asking people to please add one or maybe an update to the mod.json spec website (https://coffin-wiki.basil.cafe/modding/mod-json) is in order..

gBasil commented on README.md:
```diff
@@ -45,9 +43,11 @@ Tags are a quick way to organise mods into various categories. **They are option
-- The `repo.json` response contains each mod in the above format, but with the individual mod's information JSON inlined into a `modJson` field (which has extra properties).
+
+## Repository API
+- Llamaware provides a single JSON file with all the mods in the repository.
```
Being pedantic here, but I don't think this should be formatted as a list.

gBasil commented:
> Not sure what to do about the icon [...] maybe an update to the mod.json spec website

A spec update, yeah. Probably best to create an issue in the wiki (i'll tag it as a `mod.json` spec proposal), iron out any of the details there, and then work on a commit/PR.

kleineluka commented:
ok, ill submit a pr to there later. hopefully this is fine now here but if you have other critiques i will implement them

gBasil commented:
Instead of asking for a bajillion tiny changes, I've gone and made them myself to save you the trouble.
